src/modules/sandbox/hooks/useUser.ts

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState } from 'react';
+import { useAppDispatch, useAppSelector } from '@common/state/hooks';
+import { selectActiveNetwork } from '@common/state/network-slice';
+import { useAddressQueries } from '@features/address/use-address-queries';
+import { useQuery } from 'react-query';
+import { transactionQK, TransactionQueryKeys } from '@features/transaction/query-keys';
+import {
+  connect,
+  disconnect,
+  selectUserData,
+  selectUserSession,
+  setUserData,
+} from '@modules/sandbox/sandbox-slice';
+import { AuthOptions } from '@stacks/connect/dist/types/types/auth';
+
+export const useUser = () => {
+  const dispatch = useAppDispatch();
+  const activeNetworkMode = useAppSelector(selectActiveNetwork).mode;
+  const userSession = useAppSelector(selectUserSession);
+  const userData = useAppSelector(selectUserData);
+  const [mounted, setMounted] = useState(false);
+  const queries = useAddressQueries();
+
+  useEffect(() => setMounted(true), []);
+
+  useEffect(() => {
+    if (userSession.isUserSignedIn()) {
+      dispatch(setUserData({ userData: userSession.loadUserData() }));
+    }
+  }, []);
```

Comment:
Does this work when the user signed in status changes? I'm wondering if the return value of `userSession.isUserSignedIn()` should maybe be watched in the dependency array of this `useEffect`

Reply:
It looks like this is working when I test it, so it's probably fine -- but there might be an edge case I'm not considering in my manual tests

Reply:
The intention of this effect is to check if a user is already logged in (after reloading/revisiting the website) and set user data, so they don't need to log in again after reloading. When a user disconnects, we set the user data to undefined in the reducer.